NBO Gamification/NBO Gamification/Networking/Encoding/URLParameterEncoder.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  URLParameterEncoder.swift
+//  NBO Gamification
+//
+//  Created by facundop on 24/11/2018.
+//  Copyright © 2018 nbogamification. All rights reserved.
+//
+
+import Foundation
+
+public struct URLParameterEncoder: ParameterEncoder {
+    
+    public static func encode(urlRequest: inout URLRequest, with parameters: Parameters) throws {
+        
+        guard let url = urlRequest.url else {throw NetworkError.missingURL}
+        
+        if var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false), !parameters.isEmpty {
+            
+            urlComponents.queryItems = [URLQueryItem]()
+            
+            for (key, value) in parameters {
+                let queryItem = URLQueryItem(
+                    name: key,
+                    value: "\(value)".addingPercentEncoding(withAllowedCharacters: .urlHostAllowed))
+                urlComponents.queryItems?.append(queryItem)
+                
+                urlComponents.queryItems?.append(queryItem)
+            }
+            
+            urlRequest.url = urlComponents.url
```

Comment:
Maybe you wanted to use url instead of urlRequest

Reply:
I used urlRequest to make it clear that I'm modifying the inout paramenter